File: ga/gpt3_5_result_2024_6/result/1968B_gpt/46_1968B.c
use std::io::{self, BufRead};

fn main() {
    let stdin = io::stdin();
    let mut input = stdin.lock();
    
    let mut input_line = String::new();
    input.read_line(&mut input_line).unwrap();
    let t: usize = input_line.trim().parse().unwrap();
    
    for _ in 0..t {
        let mut input_line = String::new();
        input.read_line(&mut input_line).unwrap();
        let mut iter = input_line.trim().split_whitespace();
        let n: usize = iter.next().unwrap().parse().unwrap();
        let m: usize = iter.next().unwrap().parse().unwrap();
        
        let mut sn = String::new();
        input.read_line(&mut sn).unwrap();
        let sn = sn.trim();
        
        let mut sm = String::new();
        input.read_line(&mut sm).unwrap();
        let sm = sm.trim();
        
        let (mut i, mut j) = (0, 0);
        while i < n && j < m {
            if sn.chars().nth(i) == sm.chars().nth(j) {
                i += 1;
            }
            j += 1;
        }
        
        println!("{}", i);
    }
}

File: ga/glm4_result_2024_6/result/1968B_glm/46_1968B.rs
use std::io;

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let t: i32 = input.trim().parse().expect("Please type a number");

    for _ in 0..t {
        input.clear();
        io::stdin().read_line(&mut input).expect("Failed to read line");
        let mut parts = input.trim().split_whitespace();
        let n: i32 = parts.next().unwrap().parse().expect("Please type a number");
        let m: i32 = parts.next().unwrap().parse().expect("Please type a number");
        let sn: String = parts.next().unwrap().to_string();
        let sm: String = parts.next().unwrap().to_string();

        let mut i = 0;
        let mut j = 0;
        while i < n as usize && j < m as usize {
            if sn.chars().nth(i) == sm.chars().nth(j) {
                i += 1;
            }
            j += 1;
        }

        println!("{}", i);
    }
}